bpf/ut/attach_test.go

```diff
@@ -56,26 +56,19 @@ func TestJumpMapCleanup(t *testing.T) {
 		// Start with a clean base state in case another test left something behind.
 		t.Log("Doing initial clean up")
 		tc.CleanUpJumpMaps()
-
 		t.Log("Adding program, should add one dir and one map.")
 		startingJumpMaps := countJumpMaps()
 		startingTCDirs := countTCDirs()
 		ap.HostIP = net.ParseIP("10.0.0.1")
 		ap.IntfIP = net.ParseIP("10.0.0.2")
 		err := tc.EnsureQdisc(ap.Iface)
 		Expect(err).NotTo(HaveOccurred())
-		err = ap.AttachProgram()
+		opts, err := ap.AttachProgram()
 		Expect(err).NotTo(HaveOccurred())
+		Expect(opts).NotTo(Equal(nil))
 		Expect(countJumpMaps()).To(BeNumerically("==", startingJumpMaps+1), "unexpected number of jump maps")
 		Expect(countTCDirs()).To(BeNumerically("==", startingTCDirs+1), "unexpected number of TC dirs")
 
-		t.Log("Replacing program should add another map and dir.")
-		ap.HostIP = net.ParseIP("10.0.0.2")
-		err = ap.AttachProgram()
```

Comment:
You should keep the code that does a second attach but remove the lines that expect 2 jump maps; there should only be one even though you reattached the program.